lib/controllers.js

```diff
@@ -517,6 +518,88 @@ module.exports = function(rootDirectory, logger, indexDocument, errorDocument) {
             deleteObjects(req, res);
           })
         );
+      } else if (
+        req.headers["content-type"].indexOf("multipart/form-data") !== -1
+      ) {
+        const fields = {
+          bucket: req.params.bucket
+        };
+        const form = new multiparty.Form();
+        let chain = Promise.resolve();
+        let fileWasFound = false;
+
+        form.on("error", err => {
+          res.status(400).send(err);
+        });
+
+        form.on("part", part => {
+          if (!part.filename) {
```

Comment:
It took me a while to realized this condition filters out the parts containing the actual file upload. Maybe add comments like `// parse request fields` in this block and `// parse file upload parts` in the `else` block?